lib/elixir/lib/calendar/time.ex

```diff
@@ -60,16 +60,49 @@ defmodule Time do
   @doc """
   Returns the current time in UTC.
 
+  You can pass a time unit to automatically truncate the resulting time.
+
+  The default unit if none gets passed is `:microsecond`.
+
   ## Examples
 
       iex> time = Time.utc_now()
       iex> time.hour >= 0
       true
 
+      iex> time = Time.utc_now(:second)
+      iex> time.microsecond
+      {0, 0}
+
   """
   @doc since: "1.4.0"
-  @spec utc_now(Calendar.calendar()) :: t
-  def utc_now(calendar \\ Calendar.ISO) do
+  @spec utc_now(Calendar.calendar() | :microsecond | :millisecond | :second) :: t
+  def utc_now(calendar_or_time_unit \\ Calendar.ISO) do
+    case calendar_or_time_unit do
+      unit when unit in [:microsecond, :millisecond, :second] ->
+        utc_now(unit, Calendar.ISO)
+
+      calendar ->
+        utc_now(:microsecond, calendar)
+    end
+  end
+
+  @doc """
+  Returns the current time in UTC, supporting a precision and a specific calendar.
+
+  ## Examples
+
+      iex> time = Time.utc_now(:microsecond, Calendar.ISO)
+      iex> time.hour >= 0
+
+      iex> time = Time.utc_now(:second, Calendar.ISO)
+      iex> time.microsecond
+      {0, 0}
+
+  """
+  @doc since: "1.9.0"
+  @spec utc_now(:microsecond | :millisecond | :second, Calendar.calendar()) :: t
+  def utc_now(time_unit, calendar) when time_unit in [:microsecond, :millisecond, :second] do
     {:ok, _, time, microsecond} = Calendar.ISO.from_unix(:os.system_time(), :native)
```

Comment:
I'd say we should pass the time unit as argument here:

```suggestion
    {:ok, _, time, microsecond} = Calendar.ISO.from_unix(:os.system_time(), time_unit)
```

and allow native as time unit, as done in NaiveDateTime.

Reply:
Added `:native`. I also switched `:os.system_time()` for `System.os_time(time_unit)`.